pkg/server/commands/read_authzmodel.go

```diff
@@ -36,16 +39,22 @@ func NewReadAuthorizationModelQuery(backend storage.AuthorizationModelReadBacken
 	return m
 }
 
-func (q *ReadAuthorizationModelQuery) Execute(ctx context.Context, req *openfgav1.ReadAuthorizationModelRequest) (*openfgav1.ReadAuthorizationModelResponse, error) {
+func (q *ReadAuthorizationModelQuery) Execute(
+	ctx context.Context,
+	req *openfgav1.ReadAuthorizationModelRequest,
+) (*openfgav1.ReadAuthorizationModelResponse, error) {
 	modelID := req.GetId()
-	azm, err := q.backend.ReadAuthorizationModel(ctx, req.GetStoreId(), modelID)
+
+	typesys, err := q.typesystemResolver.ResolveTypesystem(ctx, req.GetStoreId(), req.GetId(), false)
```

Comment:
This is changing the behavior of the API, because `ResolveTypesystem` is also doing model validation, whereas `backend.ReadAuthorizationModel` is not

Reply:
Model validation is skipped.

See `, false)` at the end of the function call.

Reply:
Ah sorry my bad.

My concern with this approach is that it's not obvious that this call is doing model validations. A `WithValidate(true)` would be more obvious

Reply:
What is it specifically that is _too much_? If it's the additional validation bit, then we can work through an alternative strategy around that. Keep in mind that the reason we have `typesystem.NewAndValidate` in the resolver is because of a historical reasoning. At some point we realized we had invalid models because of a bug we found in a prior release, and so we added validation to the model resolver so that we can holistically inherit that functionality whenever and wherever we resolved a model.

The typesystem resolver _may_ be doing too much, and you may be right there, but don't forget the reasoning behind why it's currently doing what it's doing. I think the typesystem resolver still stands as an abstraction. We have a `storeID` and `modelID` and we want to resolve to a `*TypeSystem`. Seems logical to have an interface/signature representing as such.